search_service/proxy/elasticsearch.py

```diff
@@ -451,3 +452,24 @@ def delete_document(self, *, data: List[str], index: str) -> str:
             return ''
 
         return self._delete_document_helper(data=data, index=index)
+
+    @timer_with_counter
+    def clean_documents(self, *, before: Optional[int], index: str) -> Dict[str, Any]:
```

Comment:
will the ES search service have multiple search documents?
just curious does square not use https://github.com/lyft/amundsendatabuilder/blob/master/databuilder/publisher/elasticsearch_publisher.py to publish search ES index?

I felt the clean up job should be handle in the ETL/databuilder side though... WDYT @alran @jinhyukchang 

cc @verdan not sure whether Atlas provides similar functionality....

Reply:
The documents referred to here are each entry that is added to a search index. So this method, when called, basically goes into your search index and finds all documents that haven't been updated in a certain amount of time and removes them. 

Separately, we actually don't populate our search via ETL. We created our own version of databuilder that runs jobs and programmatically imports data to elasticsearch (creating documents from the data) using `create_document()` in the searchlibrary.